server/middleware/auth.ts

```diff
@@ -0,0 +1,29 @@
+import { RequestHandler } from 'express';
+import { UnauthorizedError } from 'express-response-errors';
+import jwt from 'jsonwebtoken';
+
+import { validateToken } from 'server/lib/auth';
+import { User } from 'server/models';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+
+export const attachUser: RequestHandler = async (req, res, next) => {
+  const token = req.cookies[AUTH_COOKIE_NAME];
+  if (validateToken(token)) {
+    const payload = jwt.decode(token) as jwt.JwtPayload;
+    if (payload.user_id) {
+      const user = await User.findByPk(payload.user_id);
+      req.user = user;
+    }
+  }
+
+  next();
+};
```

Comment:
This kinda does what you'd expect passport to do - where if the user is logged in, it will check the cookies for a valid token, look up the user from our DB, then attach that user to `req.user`.  No need to figure out serialization/deserialization here!  Though this is definitely the "deserialization" part (the serialization happens when we sign the token when they login)